source/extensions/filters/network/dubbo_proxy/decoder_event_handler.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+
+#include "common/buffer/buffer_impl.h"
+
+#include "extensions/filters/network/dubbo_proxy/message.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace DubboProxy {
+
+/**
+ * This class provides the pass-through capability of the original data of
+ * the Dubbo protocol to improve the forwarding efficiency
+ * when no modification of the original data is required.
+ */
+class ProtocolDataPassthroughConverter {
+public:
+  ProtocolDataPassthroughConverter() {}
+  virtual ~ProtocolDataPassthroughConverter() {}
+
+  void initProtocolConverter(Buffer::Instance& buffer) { buffer_ = &buffer; }
+
+  /**
+   * Transfer the original header data of the Dubbo protocol,
+   * it's called after the protocol's header data is parsed.
+   * @param header_buf raw header data
+   * @param size The size of the head.
+   */
+  virtual Network::FilterStatus transferHeaderTo(const Buffer::Instance& header_buf, size_t size) {
+    if (buffer_ != nullptr) {
+      Buffer::OwnedImpl copy(header_buf);
+      buffer_->move(copy, size);
+    }
+    return Network::FilterStatus::Continue;
+  }
+
+  /**
+   * Transfer the original body data of the Dubbo protocol
+   * it's called after the protocol's body data is parsed.
+   * @param header_buf raw body data
+   * @param size The size of the body.
+   */
+  virtual Network::FilterStatus transferBodyTo(const Buffer::Instance& body_buf, size_t size) {
+    if (buffer_ != nullptr) {
+      Buffer::OwnedImpl copy(body_buf);
```

Comment:
ditto

Reply:
Users can define multiple Dubbo filters, and some filters will not get data if we use move.